### PR TITLE
[NBS Docs] Added SingIn Resolvers Config Details

### PR DESCRIPTION
**Pull request by awanlin**
## Hey, I just made a Pull Request!

This adds the `signIn.resolvers` details to the auth docs for the following providers:

- Atlassian
- GCP IAP
- GitHub
- GitLab
- Google
- Microsoft
- oauth2 Proxy
- Okta
- VMware Cloud

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/auth/atlassian/provider.md [hunk 1]
```diff
@@ -47,8 +47,15 @@ auth:
         clientId: ${AUTH_ATLASSIAN_CLIENT_ID}
         clientSecret: ${AUTH_ATLASSIAN_CLIENT_SECRET}
         scope: ${AUTH_ATLASSIAN_SCOPES}
+        signIn:
+          resolvers:
```
Comment (camilaibs) [inline docs/auth/atlassian/provider.md:51]: Some open questions that I would expect to find answer on this doc page:
- Are these all resolvers available? If not where can I find a comprehensive list?
- Can I also create custom ones and use them in this provider?

File: docs/auth/google/gcp-iap-auth.md [hunk 2]
```diff
@@ -94,6 +101,27 @@ Now the backend is ready to serve auth requests on the
 `/api/auth/gcp-iap/refresh` endpoint. All that's left is to update the frontend
 sign-in mechanism to poll that endpoint through the IAP, on the user's behalf.
 
+### New Backend System
```
Comment (camilaibs) [inline docs/auth/google/gcp-iap-auth.md:104]: Would this be a good time to restructure the page to include only the "new" way of installing the plugin and module and create an "old" page with the content for people who are using the legacy backend system?

**Review by camilaibs — APPROVED** [review]
Thank you for writing down these more detailed examples, they are so valuable 🎉 . It would be great to know where to find all resolvers available and how to create custom ones in case you find time help us adding that information as well (It doesn't block us from merging the improvements you already did, just something I would find nice to have on these docs) 🙂 .

File: docs/auth/atlassian/provider.md (hunk 1 above)
Comment (awanlin) [reply docs/auth/atlassian/provider.md:51]: I've added a dedicated resolvers section to each of these to address these two items 👍

File: docs/auth/google/gcp-iap-auth.md (hunk 2 above)
Comment (awanlin) [reply docs/auth/google/gcp-iap-auth.md]: After thinking this over I think in this case as it's just a small section it doesn't make sense to have a full "old" page. what I did instead was invert this so the new backend system section is the default and the old backend system is below that.

**Review by benjdlambert — COMMENTED** [review]
Looks good to me, would like some eyes from @Rugvip / @freben also, as both them and @camilaibs know more than me on the auth parts.

Especially the `oauth2proxy` question 😅

File: docs/auth/oauth2-proxy/provider.md [hunk 3]
```diff
@@ -22,44 +22,26 @@ The provider configuration can be added to your `app-config.yaml` under the root
 auth:
   environment: development
   providers:
-    oauth2Proxy: {}
+    oauth2Proxy:
```
[the remark this answers is not in the document]
Comment (Rugvip) [reply docs/auth/oauth2-proxy/provider.md:25]: Yep looks good 👍

File: docs/auth/atlassian/provider.md [hunk 4]
```diff
@@ -47,6 +47,11 @@ auth:
...
+            - resolver: emailMatchingUserEntityProfileEmail
+            - resolver: emailLocalPartMatchingUserEntityName
+            - resolver: usernameMatchingUserEntityName
```
Comment (Rugvip) [inline docs/auth/atlassian/provider.md:54]: I'm feeling there's a risk that some people might copy this verbatim, which is never what you'd want. Do you agree and any ideas for a different way to present these?

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! 👍 🎉 

All looks good but with a worry about the way the YAML examples are presented that I think might be worth iterating on a bit

File: docs/auth/atlassian/provider.md (hunk 4 above)
Comment (awanlin) [reply docs/auth/atlassian/provider.md:54]: They will for sure copy this as is, I've seen that happen with the migration guide already when auth issues come up on Discord. I didn't realize that having them like this was problematic?

File: docs/auth/oauth2-proxy/provider.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (awanlin) [reply docs/auth/oauth2-proxy/provider.md:25]: Nice!

File: docs/auth/atlassian/provider.md (hunk 4 above)
Comment (awanlin) [reply docs/auth/atlassian/provider.md:55]: I've updated each with this comment: `# typically you would pick one of these`

**Review by Rugvip — APPROVED** [review]
:ok_hand:, :shipit:

**Comment by awanlin** [comment]
Thanks for the reviews @camilaibs @benjdlambert and @Rugvip for getting this merged! 🚀

**Pull request merged**